src/utils.jl

```diff
@@ -73,6 +75,8 @@ dim(x::RowVecs) = size(x.X, 2)
 
 pairwise(d::PreMetric, x::RowVecs) = Distances.pairwise(d, x.X; dims=1)
 pairwise(d::PreMetric, x::RowVecs, y::RowVecs) = Distances.pairwise(d, x.X, y.X; dims=1)
+pairwise(d::PreMetric, x::AbstractVector, y::RowVecs) = Distances.pairwise(d, transpose(hcat(x...)), y.X; dims=1)
+pairwise(d::PreMetric, x::RowVecs, y::AbstractVector) = Distances.pairwise(d, x.X, transpose(hcat(y...)); dims=1)
```

Comment:
```suggestion
function pairwise(d::PreMetric, x::AbstractVector, y::RowVecs)
    return Distances.pairwise(d, permutedims(reduce(hcat, x)), y.X; dims=1)
end
function pairwise(d::PreMetric, x::RowVecs, y::AbstractVector)
    return Distances.pairwise(d, x.X, permutedims(reduce(hcat, y)); dims=1)
end
```
Alternatively, one could fall back to the generic implementation in this case:
```suggestion
pairwise(d::PreMetric, x::AbstractVector, y::RowVecs) = pairwise(d, x, collect(y))
pairwise(d::PreMetric, x::RowVecs, y::AbstractVector) = pairwise(d, collect(x), y)
```